### PR TITLE
avoid run black flake8 and bandit for each python version and platform

### PR DESCRIPTION
**Pull request by ernestoarbitrio**
## Description

Avoid black flake8 and bandit task to be ran in each matrix step. This should reduce the runtime of the CI

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Fixes: 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/histolab/histolab/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -35,6 +34,39 @@ jobs:
...
+        with:
+          python-version: "3.10"
+
+      - name: Install dependencies
+        id: python-dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install flake8==3.8.4
+          python -m pip install black==22.3.0
+          python -m pip install bandit==1.7.0
```
Comment (patrick91) [inline .github/workflows/tests.yml:53]: could you use pre-commit for this instead?

Comment (ernestoarbitrio) [reply .github/workflows/tests.yml:53]: What do you mean? Run precommit in the ci instead of installing the pinned versions of those packages?

Comment (patrick91) [reply .github/workflows/tests.yml:53]: yes, exactly, or use https://pre-commit.ci/ (its free for open source) :)

Comment (ernestoarbitrio) [reply .github/workflows/tests.yml]: done, thanks for the suggestion :)

**Review by patrick91 — APPROVED** [review]

**Pull request merged**